### DISCUSSION TITLE
How to update multiple rows at a time in hasura cloud

### DISCUSSION THREAD

**Discussion by Tejesh-Duvvuru** [hasura/graphql-engine#9444] [Question]
I am writing mutation but it throwing error is 
![image](https://user-images.githubusercontent.com/101654993/219625747-27dc139b-485f-49a2-8bac-ea3f69d318ac.png)

mutation userFcmTokenTable($objects: [customprograms_set_input!]!) {
  update_customprograms_many(updates: {where: {id: {_in: ["124567", "124568"]}}, _set: $objects}) {
    affected_rows
    returning {
      id
      emailId
    }
  }
}

-----------------------------------------------------------------------------------
passing variables like this

{
  "objects": [
    {
      "id": "124567",
      "name": "Program 1",
      "duration_in_min": 60,
      "metadata": {
        "key1": "value1",
        "key2": "value2"
      }
    },
    {
      "id": "124568",
      "name": "Program 2",
      "duration_in_min": 90,
      "metadata": {
        "key3": "value3",
        "key4": "value4"
      }
    }
  ]
}

**Comment by meetzaveri** [comment]
Hey @Tejesh-Duvvuru 

Have you verified the fields which are required for `customprograms_set_input` type ? The object (containing fields) you are passing , should match  with `customprograms_set_input` type.

You can check these types by clicking on "Docs" button present in API explorer (on the extreme right side). Then click on `mutation_root` and find your mutation's type schema in it. Have attached screenshots for visual aid.

<p float="left">
  <img src="https://user-images.githubusercontent.com/21293048/220086945-8c98f1a4-15de-4714-9bfc-99cdacca0afb.png" width="400" />
  <img src="https://user-images.githubusercontent.com/21293048/220086914-70a22643-b59d-4d0e-9c84-6a8c34c63e24.png" width="400" /> 
</p>

**Reply by Tejesh-Duvvuru** [reply]
But I am also doing same it is not working
